src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -63,27 +65,124 @@ public class PermalinkService : IPermalinkService
             _tableBuilderService = tableBuilderService;
             _permalinkCsvMetaService = permalinkCsvMetaService;
             _blobStorageService = blobStorageService;
+            _frontendService = frontendService;
             _subjectRepository = subjectRepository;
             _publicationRepository = publicationRepository;
             _releaseRepository = releaseRepository;
             _mapper = mapper;
         }
 
+        public async Task<Either<ActionResult, PermalinkViewModel>> GetPermalink(Guid permalinkId,
+            CancellationToken cancellationToken = default)
+        {
+            return await Find(permalinkId, cancellationToken).OnSuccess(BuildViewModel);
+        }
+
+        private async Task<Either<ActionResult, Permalink>> Find(Guid permalinkId,
+            CancellationToken cancellationToken)
+        {
+            return await _contentDbContext.Permalinks
+                .SingleOrNotFoundAsync(
+                    predicate: permalink => permalink.Id == permalinkId &&
+                                            (!permalink.Legacy || permalink.LegacyHasSnapshot == true),
+                    cancellationToken: cancellationToken)
+                .OnSuccess<ActionResult, Permalink, Permalink>(async permalink =>
+                {
+                    try
+                    {
+                        var tableJson = await _blobStorageService.DownloadBlobText(
+                            containerName: BlobContainers.PermalinkSnapshots,
+                            path: $"{permalink.Id}.json",
+                            cancellationToken: cancellationToken);
+
+                        permalink.Table = JsonConvert.DeserializeObject<dynamic>(tableJson);
+                        return permalink;
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        return new NotFoundResult();
+                    }
+                });
```

Comment:
This is a little inconsistent with `GetLegacy` where this has been extracted out into a separate private method. Should we do something similar here?

Reply:
I think I abstracted it out to make it easier to reuse in #4001, so we can do the same here, yep.